File: frontend/src/types/patent.ts
export interface Patent {
  id: number;

  application_id: string | null;
  application_url: string | null;

  application_number: string | null;
  application_date: string | null;

  country: string | null;

  title: string | null;

  ipc: string | null;
}

export interface PatentListResponse {
  items: Patent[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}

File: backend/app/models/patent.py
from sqlalchemy import Column, Integer, String, Text

from app.database import Base


class Patent(Base):
    __tablename__ = "patents"

    id = Column(Integer, primary_key=True, index=True)

    application_id = Column(String(100), index=True)

    application_url = Column(Text)

    application_number = Column(String(255), index=True)

    application_date = Column(String(100))

    country = Column(String(100), index=True)

    title = Column(Text)

    ipc = Column(Text)